python/GafferUI/VectorDataWidget.py

```diff
@@ -204,8 +206,9 @@ def setData( self, data ) :
 		# by self.__removeSelection.
 
 		if data is not None :
-			if not isinstance( data, list ) :
-				data = [ data ]
+			if not isinstance( data, dict ) :
```

Comment:
This is obviously gonna break every single other user of VectorDataWidget. Looks like the main motivation is to get the headings you want? If so, it is already possible to pass headings as a list via the `header` constructor parameter. Let's go one way or the other, and make sure we don't break other code.

As I noted on `setColumnToolTips()`, it is a bit rubbish that different aspects of the model are specified in different places, so they can get out of sync. I do like the fact that a dictionary keeps the header and columns in a single structure, and had I done that from the start that might have been better. I'm not sure how we'd extend the dict approach to the tooltips as well though, without introducing a public Column class of some sort.

Thoughts? I think it's either charge forwards and switch things up across the board, or use the `header` stuff we already have (with any tweaks needed).